modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/common/main/infrastructure/ApiClientBase.kt.mustache

```diff
@@ -46,9 +45,8 @@ public open class ApiClientBase {
         }
     }
 
-    internal constructor(baseUrl: String, client: HttpClient, serializer: KotlinxSerializer) {
+    internal constructor(baseUrl: String, client: HttpClient) {
         this.baseUrl = baseUrl
-        this.serializer = serializer
         this.client = client
     }
```

Comment:
So I've read the code and we basically seem to need to get this passed, this will probably go away once I rewrite the request logic to be less generic, but until then I think we need this

Reply:
I'm not sure why is it required since assuming ContentType header is set and models are annotated correctly for kotlinx.serialization - Ktor can do it's magic and we don't have to filter-our the ContentType header.
Regardless, having the option to provide a custom HttpClient is essential.

Reply:
If the serialization still works even though we use the rather weird logic it's fine

Reply:
As said before I'd like most of the `ApiClientBase` stuff go away anyway, so it's probably fine

Reply:
done